src/components/LeftSidebar/LeftSidebar.astro

```diff
@@ -1,28 +1,29 @@
 ---
-import { SIDEBAR, MENU, Frontmatter } from "../../config"
+import { Sections } from "~/content/config"
+import { SIDEBAR, MENU, SectionContent } from "../../config"
 import { flattenChildren } from "../../scripts/flatten-children"
 export type Props = {
   currentPage: string
-  content: Frontmatter
+  section?: Sections
 }
-const { currentPage, content } = Astro.props as Props
+const { currentPage, section } = Astro.props
 
 // SIDEBAR is a flat array. Group it by sections to properly render.
-
-function getSidebarSections(section: string) {
-  if (section === "global") {
+const getSidebarSections = (section?: Sections) => {
+  if (!section) {
+    return []
+  } else if (section === "global") {
     return Object.keys(SIDEBAR).flatMap((parentSection) => {
-      return SIDEBAR[parentSection].map((entry) => ({ ...entry, parentSection }))
+      return SIDEBAR[parentSection].map((entry: any) => ({ ...entry, parentSection }))
```

Comment:
```suggestion
      return SIDEBAR[parentSection].map((entry: keyof typeof SIDEBAR) => ({ ...entry, parentSection }))
```

Reply:
I haven't tested out this type, but I think the correct type should be somewhat similar to that